packages/main/src/Menu.ts

```diff
@@ -437,26 +451,29 @@ class Menu extends UI5Element {
 	_createSubMenu(item: MenuItem, openerId: string) {
 		const ctor = this.constructor as typeof Menu;
 		const subMenu = document.createElement(ctor.getMetadata().getTag()) as Menu;
-		const fragment = document.createDocumentFragment();
 
 		subMenu._isSubMenu = true;
 		subMenu.setAttribute("id", `submenu-${openerId}`);
 		subMenu._parentMenuItem = item;
 		subMenu.busy = item.busy;
 		subMenu.busyDelay = item.busyDelay;
-		const subItems = item.children;
-		let clonedItem,
-			idx;
-		for (idx = 0; idx < subItems.length; idx++) {
-			clonedItem = subItems[idx].cloneNode(true);
-			fragment.appendChild(clonedItem);
-		}
-		subMenu.appendChild(fragment);
+		this._cloneItems(item, subMenu);
 		this.staticAreaItem!.shadowRoot!.querySelector(".ui5-menu-submenus")!.appendChild(subMenu);
 		item._subMenu = subMenu;
 	}
 
+	_cloneItems(item: MenuItem, menu: Menu) {
+		for (let i = 0; i < item.items.length; ++i) {
+			const clonedItem = item.items[i].cloneNode(true);
+			menu.appendChild(clonedItem);
```

Comment:
When we append elements to the DOM especially in loops, it is recommended to use the document fragment and first add all elements to the fragment and then the fragment to the DOM - one reflow, otherwise one reflow per appendChild call.